src/Lib.P2P.Tests/MultiAdressBlackListTest.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MultiFormats;
+
+namespace Lib.P2P.Tests
+{
+    [TestClass]
+    public class MultiAddressBlackListTest
+    {
+        private MultiAddress a = "/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3";
+        private MultiAddress a1 = "/ip4/127.0.0.1/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3";
+        private MultiAddress b = "/p2p/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3";
+        private MultiAddress c = "/ipfs/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64";
+        private MultiAddress d = "/p2p/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64";
+
+        [TestMethod]
+        public void Allowed()
+        {
+            var policy = new MultiAddressBlackList();
+            policy.Add(a);
+            policy.Add(b);
+            Assert.IsFalse(policy.IsAllowed(a));
+            Assert.IsFalse(policy.IsAllowed(a1));
+            Assert.IsFalse(policy.IsAllowed(b));
+            Assert.IsTrue(policy.IsAllowed(c));
+            Assert.IsTrue(policy.IsAllowed(d));
+        }
+
+        [TestMethod]
+        public void Allowed_Alias()
+        {
+            var policy = new MultiAddressBlackList();
+            policy.Add(a);
+            Assert.IsFalse(policy.IsAllowed(a));
+            Assert.IsFalse(policy.IsAllowed(a1));
+            Assert.IsFalse(policy.IsAllowed(b));
+            Assert.IsTrue(policy.IsAllowed(c));
+            Assert.IsTrue(policy.IsAllowed(d));
+        }
+
+        [TestMethod]
+        public void Empty()
+        {
+            var policy = new MultiAddressBlackList();
+            Assert.IsTrue(policy.IsAllowed(a));
+        }
+
+        [TestMethod]
+        public void Collection()
+        {
+            MultiAddress a = "/ip4/127.0.0.1";
+            MultiAddress b = "/ip4/127.0.0.2";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename 'b' which hides the field with the same name.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)